cg/apps/lims/api.py

```diff
@@ -85,74 +85,114 @@ def _export_artifact(self, lims_artifact):
 
     def get_received_date(self, lims_id: str) -> str:
         """Get the date when a sample was received."""
-        lims_artifacts = self.get_artifacts(process_type='CG002 - Reception Control',
-                                            samplelimsid=lims_id)
-        for artifact in lims_artifacts:
-            udf_key = 'date arrived at clinical genomics'
-            if artifact.parent_process and artifact.parent_process.udf.get(udf_key):
-                return artifact.parent_process.udf.get(udf_key)
+
+        step_names_udfs = MASTER_STEPS_UDFS['received_step']
+        received_dates = []
+
+        for process_type in step_names_udfs:
+            artifacts = self.get_artifacts(process_type=process_type, samplelimsid=lims_id)
+
+            for artifact in artifacts:
+                udf_key = step_names_udfs[process_type]
+                if artifact.parent_process and artifact.parent_process.udf.get(udf_key):
+                    received_dates.append((artifact.parent_process.date_run,
+                                           artifact.parent_process.udf.get(udf_key)))
+
+        sorted_dates = sorted(received_dates, key=lambda x: x[0])
+
+        return sorted_dates[-1][1] if sorted_dates else None
```

Comment:
why do we use the last one ([-1]) and not sort the dates the other way around and take the first one ([0])?